pkg/ddl/executor.go

```diff
@@ -1565,43 +1596,33 @@ func isIgnorableSpec(tp ast.AlterTableType) bool {
 }
 
 // GetCharsetAndCollateInTableOption will iterate the charset and collate in the options,
-// and returns the last charset and collate in options. If there is no charset in the options,
-// the returns charset will be "", the same as collate.
-func GetCharsetAndCollateInTableOption(startIdx int, options []*ast.TableOption, defaultUTF8MB4Coll string) (chs, coll string, err error) {
-	for i := startIdx; i < len(options); i++ {
-		opt := options[i]
-		// we set the charset to the last option. example: alter table t charset latin1 charset utf8 collate utf8_bin;
-		// the charset will be utf8, collate will be utf8_bin
+// and returns the last charset (chs) and collation (coll) in options.
```

Comment:
Maybe also mention in comments that `GetCharsetAndCollateInTableOption` should be used with a following `ResolveCharsetCollation` to fill in the not specified charset / collation